text/0000-ques-in-main.md

```diff
@@ -0,0 +1,514 @@
+probably better off printing diagnostics by hand, as is done now.
+
+## Unix-specific refinements
+[unix-specific-refinements]: #unix-specific-refinements
+
+The C standard only specifies `0`, `EXIT_SUCCESS` and `EXIT_FAILURE`
+as arguments to the [`exit`][exit.3] primitive.  (`EXIT_SUCCESS` is
+not guaranteed to have the value 0, but calling `exit(0)` *is*
+guaranteed to have the same effect as calling `exit(EXIT_SUCCESS)`;
+several versions of the `exit` manpage are incorrect on this point.)
+Any other argument has an implementation-defined effect.
+
+Within the Unix ecosystem, `exit` is relied upon to pass values in the
+range 0 through 127 (*not* 255) up to the parent process.  There is no
+general agreement on the meaning of specific nonzero exit codes, but
+there are many contexts that give specific codes a meaning, such as:
+
+* POSIX reserves status 127 for certain internal failures in `system`
+  and `posix_spawn` that cannot practically be reported via `errno`.
+
+* [`grep`][grep.1] is specified to exit with status 0 if it found a
+   match, 1 if it found no matches, and an unspecified value greater
+   than 1 if an error occurred.
+
+* [Automake's support for basic testing][automake-tests] defines status
+  77 to mean "test skipped" and status 99 to mean "hard error"
+  (I'm not sure precisely what "hard error" is for, but probably
+  something like "an error so severe that the entire test run should
+  be abandoned").
+
+* The BSDs have, for a long time, provided a header
+  [`sysexits.h`][sysexits] that defines a fairly rich set of exit
+  codes for general use, but as far as I know these have never seen
+  wide adoption.  Of these, `EX_USAGE` (code 64) for command-line
+  syntax errors is probably the most useful.
+
+* Rust itself uses status 101 for `panic!`, although arguably it
+  should be calling `abort` instead.  (`abort` on Unix systems sends a
+  different status to the parent than any value you can pass to `exit`.)
+
+It is unnecessary to put most of this stuff into the Rust stdlib,
+especially as some of these conventions contradict each other.
+However, the stdlib should not get in the way of a program that
+intends to conform to any of the above.  This can be done with the
+following refinements:
+
+* The stdlib provides `EXIT_SUCCESS` and `EXIT_FAILURE` constants in
+  `std::process` (since `exit` is already there).  These constants do
+  _not_ necessarily have the values that the platform's C library
+  gives them.  `EXIT_SUCCESS` is always 0.  `EXIT_FAILURE` has the same
+  value as the C library gives it, _unless_ the C library gives it the
+  value 1, in which case 2 is used instead.
+
+* All of the impls of `Termination` in the stdlib are guaranteed to
+  use only `EXIT_SUCCESS` and `EXIT_FAILURE`, with one exception:
+
+* There is a type, provisionally called `TermStatus`, which is a
+  newtype over `i32`; on Unix (but not on Windows), creating one from
```

Comment:
I'm not sure what you are meaning to get at with this?  The (but not on windows) parenthetical is because Windows does _not_ truncate the value passed to `ExitProcess`, so it is correct to create `TermStatus` quantities from arbitrary i32's on Windows.  (Technically, POSIX-compliant systems are *supposed* to make the full value passed to `_exit` available to a parent that retrieves it with `waitid` instead of `waitpid`, but I'm not aware of any actual Unix that meets that requirement.)